src/components/LandmarkDrawer.tsx

```diff
@@ -79,22 +92,93 @@ function Editor() {
   </>;
 }
 
-function LandmarkAssessmentHistory() {
-  return <Table>
+function LandmarkAssessmentHistory({ landmark } : {
+  landmark: Landmark;
+}) {  
+  const [user] = useUserContext();
+  const { data: assessments } = trpcNext.map.listLandmarkAssessment.useQuery({
+    userId: user.id,
+    landmark: landmark.名称,
+  });
+
+  const [selectedAssessment, setSelectedAssessment] = useState<LandmarkAssessment>();
+  const handleSelectAssessment = (assessment: {
+    createdAt: string;
+    score: number;
+    markdown: string | null;
+  }) => {
+    const newAssessment: LandmarkAssessment = {
+      ...assessment,
+      createdAt: new Date(assessment.createdAt) 
+    };
+    setSelectedAssessment(newAssessment);
+  };
```

Comment:
Why do you need this logic? Why can't simply call `setSelectedAssessment(assessment)`?

Reply:
Type string is not assignable to type Date:
<img width="1138" alt="Screenshot 2024-09-30 at 17 08 04" src="https://github.com/user-attachments/assets/868a5db7-7db3-4eb6-aad6-f4ec53ce14f7">

Reply:
It's a known problem. Use z.coerce.string().optional() as the type of zLandmarkAssessment.score and modify all impacted code. Refer to other places where coerce is used, for example, at src/shared/ChatMessage.ts:8